c5g7/openmc/rodded-A/build-xml-rodded-A.py

```diff
@@ -55,29 +52,26 @@
 # Add Core lattice to Core cell
 cells['Core'].fill = lattices['Core']
 
-# Instantiate a Geometry and register the root Universe
+# Instantiate a Geometry, register the root Universe, and export to XML
 geometry = openmc.Geometry()
 geometry.root_universe = universes['Root']
-
-# Instantiate a GeometryFile, register Geometry, and export to XML
-geometry_file = openmc.GeometryFile()
-geometry_file.geometry = geometry
-geometry_file.export_to_xml()
+geometry.export_to_xml()
 
 
 ###############################################################################
 #                   Exporting to OpenMC settings.xml File
 ###############################################################################
 
-# Instantiate a SettingsFile, set all runtime parameters, and export to XML
-settings_file = openmc.SettingsFile()
+# Instantiate a Settings, set all runtime parameters, and export to XML
+settings_file = openmc.Settings()
 settings_file.energy_mode = "multi-group"
-settings_file.cross_sections = "./mg_cross_sections.xml"
+settings_file.cross_sections = "./mgxs.xml"
 settings_file.batches = batches
 settings_file.inactive = inactive
 settings_file.particles = particles
 settings_file.output = {'tallies': True, 'summary': True}
-settings_file.source = Source(space=Box([-32.13, -10.71, -107.1], [10.71, 32.13, 85.68]))
+settings_file.source = openmc.Source(space=openmc.stats.Box(
+    [-32.13, -10.71, -107.1], [10.71, 32.13, 85.68], only_fissionable=True))
 settings_file.entropy_lower_left = [-32.13, -32.13, -107.1]
 settings_file.entropy_upper_right = [32.13,  32.13,  107.1]
 settings_file.entropy_dimension = [51, 51, 30]
```

Comment:
Similarly, the entropy dimensions need to change as well for this problem and rodded-B & the unrodded case:
```
 settings_file.entropy_lower_left = [-32.13, -32.13, -32.13]
 settings_file.entropy_upper_right = [32.13,  32.13,  32.13]
 settings_file.entropy_dimension = [51, 51, 51]
```

I chose 51 for the z-dimension of the mesh simply because the z mesh extent is the same size as the x-y mesh extent  You could realistically go less since it will be a more smoothly varying function in the z-direction, but, eh whatever.